components/datetime/src/format/datetime.rs

```diff
@@ -99,45 +103,65 @@ where
 #[derive(Debug, PartialEq, Copy, Clone, displaydoc::Display)]
 /// Error for `TryWriteable` implementations
 pub enum DateTimeWriteError {
-    // Data not loaded
-    /// Missing FixedDecimalFormatter
-    #[displaydoc("FixedDecimalFormatter not loaded")]
-    MissingFixedDecimalFormatter,
-    /// Missing OrdinalRules
-    #[displaydoc("OrdinalRules not loaded")]
-    MissingOrdinalRules,
-    /// Missing WeekCalculator
-    #[displaydoc("WeekCalculator not loaded")]
-    MissingWeekCalculator,
-    /// TODO
-    #[displaydoc("Names for {0:?} not loaded")]
-    MissingNames(Field),
-
-    // Something not found in data
-    // TODO: Are these actionable? Can clients even invent their own months and days?
-    /// Missing month symbol
-    #[displaydoc("Cannot find symbol for month {0:?}")]
-    MissingMonthSymbol(MonthCode),
-    /// Missing era symbol
-    #[displaydoc("Cannot find symbol for era {0:?}")]
-    MissingEraSymbol(FormattingEra),
-    /// Missing weekday symbol
-    #[displaydoc("Cannot find symbol for weekday {0:?}")]
-    MissingWeekdaySymbol(IsoWeekday),
-
-    // Invalid input
-    /// Incomplete input
+    /// An input field (such as "hour" or "month" is missing).
+    ///
+    /// This is guaranteed not to happen for `icu::calendar` inputs, but may happen for custom inputs.
```

Comment:
```suggestion
    /// The formatter requires an input field (such as IsoHour or YearInfo) that is not in the input.
    ///
    /// This *only* happens if the formatter has been created using
    /// [`TypedDateTimeNames::with_pattern`], and the pattern requires fields
    /// that are not returned by the input type.
```

Although "only happens" is a lie; we should fix it because data can trigger it